extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpRecorder.java

```diff
@@ -80,36 +82,50 @@ public void installBatchSpanProcessorForOtlp(
+    private SpanExporter createOtlpGrpcSpanExporter(OtlpExporterRuntimeConfig exporterRuntimeConfig, String endpoint,
+            Vertx vertx) {
+
+        if (exporterRuntimeConfig.traces().protocol().isPresent()) {
+            if (!exporterRuntimeConfig.traces().protocol().get().equals(HTTP_PROTOBUF)) {
+                throw new IllegalStateException("Only the GRPC Exporter is currently supported. " +
+                        "Please check `quarkus.otel.exporter.otlp.traces.protocol` property");
+            }
+        }
+
+        boolean compressionEnabled = false;
+        if (exporterRuntimeConfig.traces().compression().isPresent()) {
+            compressionEnabled = (exporterRuntimeConfig.traces().compression().get() == CompressionType.GZIP);
+        }
 
         // FIXME TLS Support. Was not available before but will be available soon.
         // exporterRuntimeConfig.traces.certificate.ifPresent(exporterBuilder::setTrustedCertificates);
         // exporterRuntimeConfig.client.ifPresent(exporterBuilder::setClientTls);
-        exporterRuntimeConfig.traces().headers().ifPresent(new Consumer<List<String>>() {
-            @Override
-            public void accept(final List<String> headers) {
-                for (Map.Entry<String, String> entry : convertKeyValueListToMap(headers).entrySet()) {
-                    String key = entry.getKey();
-                    String value = entry.getValue();
-                    exporterBuilder.addHeader(key, value);
+
+        Map<String, String> headersMap = new HashMap<>();
+        if (exporterRuntimeConfig.traces().headers().isPresent()) {
+            List<String> headers = exporterRuntimeConfig.traces().headers().get();
+            if (!headers.isEmpty()) {
+                for (String header : headers) {
+                    if (header.isEmpty()) {
+                        continue;
+                    }
+                    String[] parts = header.split("=", 2);
+                    String key = parts[0].trim();
+                    String value = parts[1].trim();
+                    headersMap.put(key, value);
                 }
             }
-        });
-        exporterRuntimeConfig.traces().compression()
-                .ifPresent(new Consumer<CompressionType>() {
-                    @Override
-                    public void accept(CompressionType compression) {
-                        exporterBuilder.setCompression(compression.getValue());
-                    }
-                });
-
-        if (!exporterRuntimeConfig.traces().protocol().orElse("").equals(HTTP_PROTOBUF)) {
-            throw new IllegalStateException("Only the GRPC Exporter is currently supported. " +
-                    "Please check `otel.exporter.otlp.traces.protocol` property");
         }
-        return exporterBuilder.build();
+
+        return new VertxGrpcExporter(
+                "otlp",
+                "span",
+                GlobalOpenTelemetry::getMeterProvider,
```

Comment:
We should avoid using `GlobalOpenTelemetry`. Since we are not using OTel Metrics yet, we can just pass in a noop `MeterProvider`